src/main/java/com/pingcap/tikv/statistics/Histogram.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pingcap.tikv.statistics;
+
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import com.pingcap.tikv.Snapshot;
+import com.pingcap.tikv.TiCluster;
+import com.pingcap.tikv.TiConfiguration;
+import com.pingcap.tikv.catalog.Catalog;
+import com.pingcap.tikv.expression.TiColumnRef;
+import com.pingcap.tikv.expression.TiConstant;
+import com.pingcap.tikv.expression.TiExpr;
+import com.pingcap.tikv.expression.scalar.Equal;
+import com.pingcap.tikv.meta.TiDBInfo;
+import com.pingcap.tikv.meta.TiIndexInfo;
+import com.pingcap.tikv.meta.TiSelectRequest;
+import com.pingcap.tikv.meta.TiTableInfo;
+import com.pingcap.tikv.operation.SchemaInfer;
+import com.pingcap.tikv.predicates.PredicateUtils;
+import com.pingcap.tikv.predicates.ScanBuilder;
+import com.pingcap.tikv.row.Row;
+import com.pingcap.tikv.types.DataTypeFactory;
+import com.pingcap.tikv.types.Types;
+import com.pingcap.tikv.util.Bucket;
+import com.pingcap.tikv.util.Comparables;
+import com.pingcap.tikv.util.RangeSplitter;
+
+import java.lang.reflect.Type;
+import java.util.Iterator;
+import java.util.List;
+
+
+public class Histogram {
+
+    private final long distinctValueOfNumber = 0;
+    public Bucket[] buckets;
+
+    private static TiConfiguration conf = TiConfiguration.createDefault(ImmutableList.of("127.0.0.1:" + 2379));
+    private static TiCluster cluster = TiCluster.getCluster(conf);
+    private static Snapshot snapshot = cluster.createSnapshot();
+
+    // histogramFromStorage from the storage to histogram.
+    public void histogramFromStorage(long tableID,long isIndex,long colID){
+        Catalog cat = cluster.getCatalog();
+        TiDBInfo db = cat.getDatabase("mysql");
```

Comment:
please do not do hard code. Instead, you can do the following:

private final String DBNAME = "mysql";